azurerm/internal/services/datafactory/data_factory_linked_service_azure_blob_storage_resource.go

```diff
@@ -71,6 +82,38 @@ func resourceDataFactoryLinkedServiceAzureBlobStorage() *schema.Resource {
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
 
+			"use_managed_identity": {
+				Type:          schema.TypeBool,
+				Optional:      true,
+				Default:       false,
+				ConflictsWith: []string{"service_principal_key", "service_principal_id"},
+				AtLeastOneOf:  []string{"service_principal_key", "service_principal_id", "use_managed_identity", "connection_string", "sas_uri"},
+			},
+
+			"service_principal_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.IsUUID,
+				RequiredWith:  []string{"service_principal_key"},
+				ConflictsWith: []string{"use_managed_identity"},
+				AtLeastOneOf:  []string{"service_principal_key", "service_principal_id", "use_managed_identity", "connection_string", "sas_uri"},
+			},
+
+			"service_principal_key": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  validation.StringIsNotEmpty,
+				RequiredWith:  []string{"service_principal_id"},
+				ConflictsWith: []string{"use_managed_identity"},
+				AtLeastOneOf:  []string{"service_principal_key", "service_principal_id", "use_managed_identity", "connection_string", "sas_uri"},
+			},
+
+			"tenant": {
```

Comment:
Assuming this is the Tenant GUID, can we rename to `tenant_id` to keep consistent with the rest of the provider?